### PR TITLE
#44 - Add image upload and copy to User Profile Edit

### PR DESCRIPTION
**Pull request by alistairshaw**
I am new to both RoR and Materials so forgive any sins.

This PR will add a bunch of functionality to the User Profile Edit form, including migrations and model updates, etc for some new fields we needed.

I've used "Active Storage" for the profile pic after doing some research on the best way to handle it. Seems to work great, although a few hoops to get the URL out and included in the payload for the front-end.

### CODE REVIEW

File: server/db/migrate/20200426015709_add_fields_to_user_profile.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class AddFieldsToUserProfile < ActiveRecord::Migration[6.0]
+  def change
+    add_column :user_profiles, :business_name, :string, :after => :user_name
+    add_column :user_profiles, :specialty, :string, :after => :user_name
+    add_column :user_profiles, :tip_url, :string, :after => :user_name
```
Comment (asleepysheepy) [inline server/db/migrate/20200426015709_add_fields_to_user_profile.rb:5]: We have a `donation_methods` table already for this. We did it as a separate table to allow users to be able to have multiple links, say to both a paypal and a venmo

File: web/looking-good/.eslintrc.js [hunk 2]
```diff
@@ -102,10 +102,6 @@ module.exports = {
     'react-hooks/rules-of-hooks': 'error',
     'react-hooks/exhaustive-deps': 'error',
 
-    'sort-imports-es6-autofix/sort-imports-es6': [2, {
-      'ignoreCase': false,
-      'ignoreMemberSort': false,
-      'memberSyntaxSortOrder': ['all', 'single', 'multiple', 'none']
-    }],
+    'sort-imports-es6-autofix/sort-imports-es6': [0],
```
Comment (asleepysheepy) [inline web/looking-good/.eslintrc.js:105]: Why did you change this?

File: web/looking-good/src/pages/user-profile-edit.js [hunk 3]
```diff
@@ -1,32 +1,50 @@
...
-  Typography,
 } from '@material-ui/core'
+import makeStyles from '@material-ui/core/styles/makeStyles'
+import MuiAlert from '@material-ui/lab/Alert'
 import { Formik } from 'formik'
-import { makeStyles } from '@material-ui/core/styles'
 import { useRouter } from 'next/router'
+import React, { useEffect, useState } from 'react'
+import FormErrors from '../components/form-errors'
+import useAuth from '../hooks/use-auth'
```
Comment (asleepysheepy) [inline web/looking-good/src/pages/user-profile-edit.js:18]: We have webpack setup to allow absolute import paths, so we should be importing from `src/components/form-errors`

File: web/looking-good/src/pages/user-profile-edit.js [hunk 4]
```diff
@@ -46,117 +64,242 @@ const UserProfileEdit = () => {
...
         })
         setIsLoading(false)
-      }).catch( () => {
+      }).catch(() => {
         setIsLoading(false)
       })
     } else {
       // If user is logged out, kick them to the sign-in
-      router.push('/login')
+      router.push('/login').then(() => {
```
Comment (asleepysheepy) [inline web/looking-good/src/pages/user-profile-edit.js:97]: The function you're passing to `.then` doesn't do anything, what's the purpose of this `.then` call?

File: web/looking-good/src/pages/user-profile-edit.js [hunk 5]
```diff
@@ -46,117 +64,242 @@ const UserProfileEdit = () => {
...
-
-      </Card>
+          </div>
+        )}
+      </Formik>
     </Container>
   )
 }
 
+// noinspection JSUnusedGlobalSymbols
```
Comment (asleepysheepy) [inline web/looking-good/src/pages/user-profile-edit.js:304]: What is this line doing?

File: web/looking-good/src/utils/theme.js [hunk 6]
```diff
@@ -5,9 +5,11 @@ import { pink } from '@material-ui/core/colors'
 const theme = createMuiTheme({
   palette: {
     primary: {
-      main: '#4300FF',
+      //main: '#4300FF',
+      main: '#2839aa',
```
Comment (asleepysheepy) [inline web/looking-good/src/utils/theme.js:9]: The colors we were using came from the Can't Stop Columbus Branding document. We probably shouldn't change them unless that document has been updated

File: web/looking-good/src/utils/theme.js [hunk 7]
```diff
@@ -5,9 +5,11 @@ import { pink } from '@material-ui/core/colors'
...
     },
     secondary: {
+      //main: '#13efc3',
```
Comment (asleepysheepy) [inline web/looking-good/src/utils/theme.js:12]: This comment doesn't need to be here

File: server/db/migrate/20200426015709_add_fields_to_user_profile.rb (hunk 1 above)
Comment (alistairshaw) [reply server/db/migrate/20200426015709_add_fields_to_user_profile.rb:5]: Yeah I saw that, and I tried to add a row to that table but was spending too much time trying to figure out RoR so I just added a field to the main profiles table for now. We can go back to the one-to-many relationship once I get a bit more ramped up on Rails and understand how to do it properly!

File: web/looking-good/.eslintrc.js (hunk 2 above)
Comment (alistairshaw) [reply web/looking-good/.eslintrc.js:105]: Oh whoops, I was having trouble getting my imports in the order the linter wanted, I intended to put this back, will do so!

File: web/looking-good/src/pages/user-profile-edit.js (hunk 3 above)
Comment (alistairshaw) [reply web/looking-good/src/pages/user-profile-edit.js:18]: Cool, I love that! My IDE put the relative paths in for me, will fix :)

File: web/looking-good/src/pages/user-profile-edit.js (hunk 4 above)
Comment (alistairshaw) [reply web/looking-good/src/pages/user-profile-edit.js:97]: No purpose, removed.

File: web/looking-good/src/pages/user-profile-edit.js (hunk 5 above)
Comment (alistairshaw) [reply web/looking-good/src/pages/user-profile-edit.js:304]: That's a Rubymine instruction for the IDE linter, removed.

File: web/looking-good/src/utils/theme.js (hunk 6 above)
Comment (alistairshaw) [reply web/looking-good/src/utils/theme.js:9]: Yeah so Rachyl told me, have swapped back to the original colors.

File: web/looking-good/src/utils/theme.js (hunk 7 above)
Comment (alistairshaw) [reply web/looking-good/src/utils/theme.js:12]: Comment removed
